pkg/services/sqlstore/permissions/dashboard.go

```diff
@@ -181,9 +180,8 @@ func (f *accessControlDashboardPermissionFilter) buildClauses() {
 	permSelector := strings.Builder{}
 	var permSelectorArgs []any
 
-	// useSelfContainedPermissions is true if the user's permissions are stored and set from the JWT token
-	// currently it's used for the extended JWT module (when the user is authenticated via a JWT token generated by Grafana)
-	useSelfContainedPermissions := f.user.IsAuthenticatedBy(login.ExtendedJWTModule)
+	// useSelfContainedPermissions is true if the signed in user's permissions are populated and the accessActionSets feature flag is not enabled
+	useSelfContainedPermissions := len(f.user.GetPermissions()) > 0 && !f.features.IsEnabledGlobally(featuremgmt.FlagAccessActionSets)
```

Comment:
I think it should be fine to use self-contained permissions if action sets are enabled as well. Or did you encounter an issue with that?

Reply:
Looking at the code a bit more, I think we can always use self-contained permissions (if it scales better than queries with permission checks). Currently we still check user's permissions even if `useSelfContainedPermissions` is set to false (we check if the user has the requested action [here](https://github.com/grafana/grafana/blob/e27e27db06c647fc760ee9b80a2d4d44a0e828e0/pkg/services/sqlstore/permissions/dashboard.go#L147)).

Reply:
I have excluded it because some action sets tests were failing: maybe they need additional setup

Reply:
Yeah, probably. I can help to look into that if we decide to go with this approach in general.